license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/AbstractLicenseMojo.java

```diff
@@ -669,7 +671,7 @@ private void executeForLicenseSet(final LicenseSet licenseSet, final Callback ca
               .map(Objects::toString).collect(Collectors.joining("\n - ")));
         }
 
-        return perDoc;
+        return new AdjustedMap<>(perDoc, perDocAdjustments);
```

Comment:
I like this hack, but it could be improved. The `perDoc` map could be this adjusted map (called differently - could be named `ForwardingMap`), where keys would be first fetched from the suppliers, otherwise from `file.name` otherwise from globalProperties maps.

there is

Reply:
`ForwardingMap` it is.

Including `file.name` with forwarding will be tricky, as it is part of the read only props passed to the adjusting call.
You'd end up with a deeply nested structure of forwarding maps, while accumulating resolver adjustments. (I tried that earlier and did not like it.)

Reply:
file.name could be hard coded or put in a map copied from the global props.